emcache/protocol.py

```diff
@@ -142,6 +143,16 @@ def data_received(self, data: bytes) -> None:
 
         self._parser.feed_data(data)
 
+    async def auth(self, username: str, password: str):
+        result = await self.auth_command(username, password)
+        if result == b"CLIENT_ERROR authentication failure":
+            raise AuthenticationError(f"Fail authentication. Incorrect username or password. Return result {result}.")
+        elif result != STORED:
+            raise AuthenticationNotSupportedError(
```

Comment:
Maybe if we just use one single error, like `AuthenticationError` will help the users to have to only catch one single exception.

Reply:
Ok